File: benchmark/src/php/trie_dict.php
<?php
$W=20000;$seed=12345;function ri($m){global$seed;$seed=($seed*16807)%2147483647;return$seed%$m;}
$nodes=1;$children=array_fill(0,26,0);$terminal=[0];
function nc(){global$nodes,$children,$terminal;for($i=0;$i<26;$i++)$children[]=0;$terminal[]=0;return$nodes++;}
$start=hrtime(true);
for($w=0;$w<$W;$w++){$l=3+ri(6);$cur=0;for($j=0;$j<$l;$j++){$c=ri(26);$idx=$cur*26+$c;if($children[$idx]==0)$children[$idx]=nc();$cur=$children[$idx];}$terminal[$cur]=1;}
$hits=0;$seed=12345;
for($w=0;$w<$W;$w++){$l=3+ri(6);$cur=0;for($j=0;$j<$l;$j++){$c=ri(26);$idx=$cur*26+$c;if($children[$idx]==0){$cur=0;break;}$cur=$children[$idx];}if($cur!=0&&$terminal[$cur]==1)$hits++;}
$seed=54321;
for($w=0;$w<$W;$w++){$l=3+ri(6);$cur=0;for($j=0;$j<$l;$j++){$c=ri(26);$idx=$cur*26+$c;if($children[$idx]==0){$cur=0;break;}$cur=$children[$idx];}if($cur!=0&&$terminal[$cur]==1)$hits++;}
$end=hrtime(true);$ms=($end-$start)/1e6;echo"Result: {$nodes}_{$hits}\n";echo"Time: ".round($ms)."ms\n";

File: benchmarks/src/ruby/trie_dict.rb
$nodes=1;$children=[0]*26;$terminal=[0]
W=20000;$seed=12345;def ri(m);$seed=($seed*16807)%2147483647;$seed%m;end
def nc;$nodes+=1;26.times{$children<<0};$terminal<<0;$nodes-1;end
start=Time.now.to_f*1000;W.times{l=3+ri(6);cur=0
 l.times{c=ri(26);idx=cur*26+c
  if $children[idx]==0;$children[idx]=nc;end
  cur=$children[idx]}
 $terminal[cur]=1}
hits=0;$seed=12345
W.times{l=3+ri(6);cur=0
 l.times{c=ri(26);idx=cur*26+c
  if $children[idx]==0;cur=0;break;end;cur=$children[idx]}
 hits+=1 if cur!=0&&$terminal[cur]==1}
$seed=54321
W.times{l=3+ri(6);cur=0
 l.times{c=ri(26);idx=cur*26+c
  if $children[idx]==0;cur=0;break;end;cur=$children[idx]}
 hits+=1 if cur!=0&&$terminal[cur]==1}
finish=Time.now.to_f*1000;puts"Result: #{$nodes}_#{hits}";puts"Time: #{(finish-start).round}ms"
